users/views.py

```diff
@@ -34,3 +35,21 @@ def get_object(self):
         return self.request.user
 
 
+class ChangePasswordView(generics.UpdateAPIView):
+    serializer_class = ChangePasswordSerializer
+    model = UserProfile
+    permission_classes = (IsAuthenticated,)
+
+    def get_object(self, queryset=None):
+        return self.request.user
+
+    def update(self, request, *args, **kwargs):
+        user = self.get_object()
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user.set_password(serializer.data.get("new_password"))
+        user.auth_token.delete()
+        token, created = Token.objects.get_or_create(user=user)
+        user.save()
```

Comment:
Thank you. It looks great.
Can you please change the order of `token, created = Token.objects.get_or_create(user=user)` with `user.save()` ? I think it is confusing. What do you think ?

After that, we can merge.